x-pack/plugins/case/common/api/connectors/mappings.ts

```diff
@@ -0,0 +1,191 @@
+});
+const GetFieldsResponseRt = rt.type({
+  defaultMappings: rt.array(ConnectorMappingsAttributesRT),
+  fields: rt.array(ConnectorFieldRt),
+});
+export type GetFieldsResponse = rt.TypeOf<typeof GetFieldsResponseRt>;
+
+export type ExternalServiceParams = Record<string, unknown>;
+
+export interface PipedField {
+  actionType: string;
+  key: string;
+  pipes: string[];
+  value: string;
+}
+export interface PrepareFieldsForTransformArgs {
+  defaultPipes: string[];
+  mappings: ConnectorMappingsAttributes[];
+  params: ServiceConnectorCaseParams;
+}
+export interface EntityInformation {
+  createdAt: string;
+  createdBy: ElasticUser;
+  updatedAt: string | null;
+  updatedBy: ElasticUser | null;
+}
+export interface TransformerArgs {
+  date?: string;
+  previousValue?: string;
+  user?: string;
+  value: string;
+}
+
+export type Transformer = (args: TransformerArgs) => TransformerArgs;
+export interface TransformFieldsArgs<P, S> {
+  currentIncident?: S;
+  fields: PipedField[];
+  params: P;
+}
+
+export const ServiceConnectorUserParams = rt.type({
+  fullName: rt.union([rt.string, rt.null]),
+  username: rt.string,
+});
+
+export const ServiceConnectorCommentParamsRt = rt.type({
+  commentId: rt.string,
+  comment: rt.string,
+  createdAt: rt.string,
+  createdBy: ServiceConnectorUserParams,
+  updatedAt: rt.union([rt.string, rt.null]),
+  updatedBy: rt.union([ServiceConnectorUserParams, rt.null]),
+});
+export const ServiceConnectorBasicCaseParamsRt = rt.type({
+  comments: rt.union([rt.array(ServiceConnectorCommentParamsRt), rt.null]),
+  createdAt: rt.string,
+  createdBy: ServiceConnectorUserParams,
+  description: rt.union([rt.string, rt.null]),
+  externalId: rt.union([rt.string, rt.null]),
+  savedObjectId: rt.string,
+  title: rt.string,
+  updatedAt: rt.union([rt.string, rt.null]),
+  updatedBy: rt.union([ServiceConnectorUserParams, rt.null]),
+});
+
+export const ConnectorPartialFieldsRt = rt.partial({
+  ...JiraFieldsRT.props,
+  ...ResilientFieldsRT.props,
+  ...ServiceNowFieldsRT.props,
+});
+
+export const ServiceConnectorCaseParamsRt = rt.intersection([
+  ServiceConnectorBasicCaseParamsRt,
+  ConnectorPartialFieldsRt,
+]);
+
+export const ServiceConnectorCaseResponseRt = rt.intersection([
+  rt.type({
+    title: rt.string,
+    id: rt.string,
+    pushedDate: rt.string,
+    url: rt.string,
+  }),
+  rt.partial({
+    comments: rt.array(
+      rt.intersection([
+        rt.type({
+          commentId: rt.string,
+          pushedDate: rt.string,
+        }),
+        rt.partial({ externalCommentId: rt.string }),
+      ])
+    ),
+  }),
+]);
+export type ServiceConnectorBasicCaseParams = rt.TypeOf<typeof ServiceConnectorBasicCaseParamsRt>;
+export type ServiceConnectorCaseParams = rt.TypeOf<typeof ServiceConnectorCaseParamsRt>;
+export type ServiceConnectorCaseResponse = rt.TypeOf<typeof ServiceConnectorCaseResponseRt>;
+export type ServiceConnectorCommentParams = rt.TypeOf<typeof ServiceConnectorCommentParamsRt>;
```

Comment:
Maybe we could move these types to `x-pack/plugins/case/common/api/connectors/index.ts` as it not mapping related.

Reply:
ok